xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-test/xwiki-platform-index-test-pageobjects/src/main/java/org/xwiki/index/test/po/RestoreDocumentConfirmationModal.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.index.test.po;
+
+import org.openqa.selenium.By;
+import org.xwiki.test.ui.po.ConfirmationModal;
+import org.xwiki.test.ui.po.ViewPage;
+
+/**
+ * Represents the actions possible on the restore document confirmation modal.
+ *
+ * @version $Id$
+ * @since 14.4RC1
+ */
+public class RestoreDocumentConfirmationModal extends ConfirmationModal
+{
+    public RestoreDocumentConfirmationModal()
+    {
+        super(By.id("restoreDocumentConfirmModal"));
+        waitUntilReady();
+    }
+
+    private void waitUntilReady()
+    {
+        getDriver().waitUntilElementIsVisible(RestoreDocumentConfirmationModal.this.container,
```

Comment:
Out of curiosity, why `RestoreDocumentConfirmationModal.this.container`? Afaics, `this.container` is enough an less verbose.

Reply:
I started with a `waitUntilCondition` and I forgot to update this part, thanks!